src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/routinghelper/WayTraversalDirection.java

```diff
@@ -0,0 +1,48 @@
+package org.openstreetmap.josm.plugins.pt_assistant.actions.routinghelper;
+
+import java.util.function.Function;
+
+import com.drew.lang.annotations.NotNull;
+import com.drew.lang.annotations.Nullable;
+import org.openstreetmap.josm.data.osm.INode;
+import org.openstreetmap.josm.data.osm.IWay;
+import org.openstreetmap.josm.data.osm.Way;
+
+public enum WayTraversalDirection {
```

Comment:
This is currently used in the transport mode to make access to the start and end nodes easier.

Instead of Way.firstNode() and Way.lastNode() you can use WayTraversalDirection.FORWARD.getStartNodeFor(way) and WayTraversalDirection.FORWARD.getEndNodeFor(way) to find the first and last node you come across.